cupy/core/core.pyx

```diff
@@ -861,34 +861,94 @@ cdef class ndarray:
             axis (int): Axis along which to sort. Default is -1, which means
                 sort along the last axis.
 
-        .. note::
-           For its implementation reason, :func:`cupy.ndarray.partition` fully
-           sorts the given array as :meth:`cupy.ndarray.sort` does. It also
-           does not support ``kind`` and ``order`` parameters that
-           :func:`numpy.partition` supports.
-
         .. seealso::
             :func:`cupy.partition` for full documentation,
             :meth:`numpy.ndarray.partition`
 
         """
-        ndim = self.ndim
+
+        if self.dtype not in [numpy.int8, numpy.uint8, numpy.int16,
+                              numpy.uint16, numpy.int32, numpy.uint32,
+                              numpy.int64, numpy.uint64, numpy.float32,
+                              numpy.float64]:
+            raise NotImplementedError('Sorting arrays with dtype \'{}\' is '
+                                      'not supported'.format(self.dtype))
+
+        cdef Py_ssize_t ndim = self.ndim
+
+        if ndim == 0:
+            raise ValueError('Sorting arrays with the rank of zero is not '
+                             'supported')
+
+        if not self._c_contiguous:
```

Comment:
Is this needed?

Reply:
Yes, the partition kernel does not support view.

Reply:
OK